app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -296,10 +298,12 @@ function FormSteps({
 
       let error: Error | undefined;
       if (isActive) {
-        element.checkValidity();
+        if (element instanceof HTMLInputElement) {
+          element.checkValidity();
+        }
 
-        if (element.validationMessage) {
-          error = new Error(element.validationMessage);
+        if ((element as HTMLInputElement).validationMessage) {
+          error = new Error((element as HTMLInputElement).validationMessage);
```

Comment:
I gave this a shot at the TS compiler seemed happy enough to only check/cast once:
```suggestion
        if (element instanceof HTMLInputElement && element.validationMessage) {
          error = new Error(element.validationMessage);
```
```suggestion
        if ((element as HTMLInputElement).validationMessage) {
          error = new Error((element as HTMLInputElement).validationMessage);
```

Reply:
> I gave this a shot at the TS compiler seemed happy enough to only check/cast once:

Good idea!